altianalysis/run_difference_with_gpao.py

```diff
@@ -25,23 +25,37 @@ def get_tile_names(folder: Path) -> List[str]:
     return filenames
 
 
-def create_one_job_one_difference(store: Store, dir_in: Path, input_file: str, output: Path):
+def create_one_job_one_difference(store: Store, dir_in: Path, second_dir: Path | None, input_file: str, output: Path):
     job_name = f"difference_{input_file}"
-    command = f"""
-docker run -t --rm --userns=host
--v {store.to_unix(dir_in)}:/input
--v {store.to_unix(output)}:/output
-ghcr.io/ignf/altianalysis:{__version__}
-python -m altianalysis.compute_difference
---dtm_lidar_file /input/{input_file}
---name_save_out /output/{input_file}
-"""
+    if second_dir is not None:
+        command = f"""
+    docker run -t --rm --userns=host --shm-size=2gb
+    -v {store.to_unix(dir_in)}:/input
+    -v {store.to_unix(second_dir)}:/second_input
+    -v {store.to_unix(output)}:/output
+    ghcr.io/ignf/altianalysis:{__version__}
+    python -m altianalysis.compute_difference
+    --dtm_lidar_file /input/{input_file}
+    --second_elevation_file /second_input/{input_file}
+    --name_save_out /output/{input_file}
+    """
+    else:
+        command = f"""
+    docker run -t --rm --userns=host --shm-size=2gb
+    -v {store.to_unix(dir_in)}:/input
+    -v {store.to_unix(output)}:/output
+    ghcr.io/ignf/altianalysis:{__version__}
+    python -m altianalysis.compute_difference
+    --dtm_lidar_file /input/{input_file}
+    --name_save_out /output/{input_file}
+    """
```

Comment:
Pour éviter la duplication de code et le risque d'oubli au moment d'une mise à jour, ça donne envie de factoriser en : 

```suggestion
   second_input_mount  = f" -v {store.to_unix(second_dir)}:/second_input" if second_dir else ""
   second_input_param = f"--second_elevation_file /second_input/{input_file}" if second_dir else ""
   
   command = f"""
    docker run -t --rm --userns=host
    -v {store.to_unix(dir_in)}:/input
    {second_input_mount}
    -v {store.to_unix(output)}:/output
    ghcr.io/ignf/altianalysis:{__version__}
    python -m altianalysis.compute_difference
    --dtm_lidar_file /input/{input_file}
    {second_input_param}
    --name_save_out /output/{input_file}
    """
```

Reply:
up @DaliCHEBBI

Reply:
remis à jour dans un autre commentaire